lightning/src/offers/invoice_request.rs

```diff
@@ -518,19 +527,35 @@ where
 
 impl UnsignedInvoiceRequest {
 	fn new(offer: &Offer, contents: InvoiceRequestContents) -> Self {
+		let mut bytes = Vec::new();
+
 		// Use the offer bytes instead of the offer TLV stream as the offer may have contained
 		// unknown TLV records, which are not stored in `OfferContents`.
-		let (payer_tlv_stream, _offer_tlv_stream, invoice_request_tlv_stream) =
-			contents.as_tlv_stream();
-		let offer_bytes = WithoutLength(&offer.bytes);
-		let unsigned_tlv_stream = (payer_tlv_stream, offer_bytes, invoice_request_tlv_stream);
+		let (
+			payer_tlv_stream, _offer_tlv_stream, invoice_request_tlv_stream,
+			_experimental_offer_tlv_stream, experimental_invoice_request_tlv_stream,
+		) = contents.as_tlv_stream();
 
-		let mut bytes = Vec::new();
-		unsigned_tlv_stream.write(&mut bytes).unwrap();
+		payer_tlv_stream.write(&mut bytes).unwrap();
 
-		let tagged_hash = TaggedHash::from_valid_tlv_stream_bytes(SIGNATURE_TAG, &bytes);
+		for record in TlvStream::new(&offer.bytes).range(OFFER_TYPES) {
+			record.write(&mut bytes).unwrap();
+		}
```

Comment:
I think we're missing test coverage that we'll include unknown offer TLVs here

Reply:
Yeah, though that predates this PR. Let me see how difficult it would be to test for this.

Reply:
Ok, so while we don't *directly* test this, I think we have sufficient enough coverage that we'll include unknown TLVs. `parses_offer_with_unknown_tlv_records` ensures we'll parse unknown TLVs, and the latest push now checks that the `Offer::bytes` match what was encoded. I did the same for parsing experimental records and added a missing tests for those in refunds.